dbmigrate-lib/src/drivers/surreal.rs

```diff
@@ -0,0 +1,142 @@
+use surreal_client::Surreal;
+use surreal_client::engine::remote::ws::{Client, Ws};
+use tokio::runtime::Runtime;
+use url::Url;
+
+use super::Driver;
+use crate::errors::{Result, ResultExt};
+
+/// The SurrealDB driver
+pub struct Surrealdb {
+    client: Surreal<Client>,
+    runtime: Runtime,
+}
+
+impl Surrealdb {
+    /// Create SurrealDB driver
+    pub fn new(url: &str) -> Result<Surrealdb> {
+        let runtime = Runtime::new().unwrap();
+
+        let parsed_url = Url::parse(url).expect(&format!("Invalid SurrealDB URL: {}", url));
```

Comment:
We need actual error handling rather than unwraps in this fn

Reply:
Done, let me know if you want to see another way to handle this

Reply:
@Keats ping :)